### ISSUE TITLE
Reorder variables per recommendations

### ISSUE THREAD

**Issue by GreatScott000** [anzwdev/al-code-outline#108]
You have added sorting of procedures, columns properties, would it be possibile to add a sort for var sections?  It should follow the variable declaration order from MS, Table, Report, Codeunit, XMLport, Page, Query, RecordRef, FieldRef(?), etc.

Once the "required" order is done it would be nice to have the remaining variables grouped by type as well.

I both cases, it would be nice to have them sorted by type name within the type.

Converting custom code often leaves the variable declarations in a non-standard order over a large number of objects/procedures and is tedious to correct manually.

**Comment by anzwdev** [comment]
Hi. Thank you for the idea, I'll take a look at it.

**Comment by anzwdev** [comment]
I've released new version of the extension, "sort variables" code action is available when cursor is in the first line of an object definition or at "var" keyword. You can also enable automatic variable sorting by adding these 2 entries to your vs code settings:
```javascript
"editor.codeActionsOnSave": {
  "source.fixAll": true
},
"alOutline.codeActionsOnSave": [
  "SortVariables"
]
```

**Comment by GreatScott000** [comment]
Thank you.  It looks like it works well.

**Comment by GreatScott000** [comment]
Once minor issue, it would appear that array[x] of Record need to be sorted with other Record variables.

**Comment by anzwdev** [comment]
Thank you for letting me know, I'll update the functionality.

**Comment by anzwdev** [comment]
I've just released a new version of the extension, array and list variables are sorted by element type now.
